frontend/src/components/commons/TreeNode.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import "../../css/lib/tree";
+import "../../css/lib/tree.css";
+import { treeNodeShape } from "../eMIB/constants";
+
+//Recursive tree building component. for each
+class TreeNode extends Component {
+  static propTypes = {
+    children: PropTypes.arrayOf(treeNodeShape).isRequired,
+    root: PropTypes.bool // optional prop. If it is present, then this is the root node
```

Comment:
I like naming booleans to that it's clear that it's a boolean: `isRoot` is more clear that it's a boolean.

Reply:
renamed in 9d1d742